resources/providers/awslib/s3/s3.go

```diff
@@ -25,8 +25,18 @@ import (
 )
 
 type BucketDescription struct {
-	Name         string
-	SSEAlgorithm string
+	Name             string
+	SSEAlgorithm     string
+	BucketPolicy     BucketPolicy
+	BucketVersioning BucketVersioning
+}
+
+// TODO: This can be better typed, but this is a complex object. See this library for example: https://github.com/liamg/iamgo/
+type BucketPolicy map[string]any
+
+type BucketVersioning struct {
```

Comment:
consider adding struct tags

Reply:
I'll fix this in a future PR as it requires changes to the policy code